subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy

```diff
@@ -789,17 +806,71 @@ tmpdir is currently ${System.getProperty("java.io.tmpdir")}""")
         }
     }
 
-    /**
-     * Gets the first problem of the given category. Assumes that there is only one problem.
-     *
-     * @return The first collected problem
-     * @throws AssertionError if there is no, or more than one, collected problem
-     * @throws IllegalStateException if the Problems API check is not enabled
-     */
-    ReceivedProblem getCollectedProblem() {
-        def problems = getCollectedProblems()
-        assert problems.size() == 1: "Expected to find exactly one problem, but found ${problems.size()} problems"
-        return problems[0]
+    static void printCollectedProblems(ReceivedProblem problem, int index) {
+        println "verifyAll(receivedProblem($index)) {"
+        println "    fqid == '${problem.definition.id.fqid}'"
+        if (problem.contextualLabel != null) {
+            println "    contextualLabel == '${problem.contextualLabel.replaceAll("'", "\\\\'")}'"
+        }
+        if (problem.details != null) {
+            println "    details == '${problem.details.replaceAll("'", "\\\\'")}'"
+        }
+        if (problem.solutions.size() == 1) {
+            println "    solutions == [ '${problem.solutions[0].replaceAll("'", "\\\\'")}' ]"
+        } else if (problem.solutions.size() > 1) {
+            println "    solutions == ["
+            problem.solutions.each { solution ->
+                println "        '${solution.replaceAll("'", "\\\\'")}',"
+            }
+            println "    ]"
+        }
+        if (problem.additionalData.size() == 1) {
+            println "    additionalData == [ '${problem.additionalData.keySet().iterator().next()}' : '${problem.additionalData.values().iterator().next()}' ]"
+        } else if (problem.additionalData.size() > 1) {
+            println "    additionalData == ["
+            problem.additionalData.each { key, value ->
+                println "        '$key' : '$value',"
+            }
+            println "    ]"
+        }
+        println "}"
+
+    }
+
+    private List<ReceivedProblem> receivedProblems
+
+    private List<ReceivedProblem> getReceivedProblems() {
+        if (receivedProblems == null) {
```

Comment:
❗ Found a bug with this approach. There are some tests which do multiple runs in one tests.
In one such case, I cannot assert the second run's problems, as the `receivedProblems` will be non-null, hence, the list of problems will be the same.
I think per-run, we should reset this collection to `null`?

Reply:
The task in question is this: `org.gradle.integtests.TaskErrorExecutionIntegrationTest#reports unknown task`, which now reports a single issue per-run

Reply:
Question here is if we should split this unique test, and/or fix the behavior in in the fixture?

Reply:
I think it doesn't matter how many Gradle runs there are in a test, the enforcement is that all of them needs to be accessed before `cleanup()` is called.  I've just fixed `TaskErrorExecutionIntegrationTest `, ptal.

Reply:
Interesting, I did the following, and got a passing test:
```
        then:
        verifyAll(receivedProblem(0)) {
            fqid == 'task-selection:no-matches'
        }
        failure.assertHasDescription("Task 'someTest' not found in root project 'test' and its subprojects. Some candidates are: 'someTask', 'someTaskA', 'someTaskB'.")
        failure.assertHasResolutions(
            GET_TASKS,
            NAME_EXPANSION,
            INFO_DEBUG,
            SCAN,
            GET_HELP
        )

        when:
        fails ":someTest"
        then:
        failure.assertHasDescription("Cannot locate tasks that match ':someTest' as task 'someTest' not found in root project 'test'. Some candidates are: 'someTask'.")
        failure.assertHasResolutions(
            GET_TASKS,
            NAME_EXPANSION,
            INFO_DEBUG,
            SCAN,
            GET_HELP
        )

        when:
        fails "a:someTest"
        then:
        failure.assertHasDescription("Cannot locate tasks that match 'a:someTest' as task 'someTest' not found in project ':a'. Some candidates are: 'someTask', 'someTaskA'.")
        failure.assertHasResolutions(
            GET_TASKS,
            NAME_EXPANSION,
            INFO_DEBUG,
            SCAN,
            GET_HELP
        )
```

I wouldn't say that this is correct, as we should have two additional runs, which should consequently fail the build

Reply:
Here is a complete example of what I would imagine as the correct behavior:
https://github.com/gradle/gradle/pull/28580/files#diff-6c30fc5dbca6d4b4a6efa7098f780be9c4ce44468be462d056964d0838464aba